### ISSUE TITLE
Using data before the request completes

### ISSUE THREAD

**Issue by oliverjam** [sofiapoh/react-dynamic-data-workshop#29]
Without fail at least half of every cohort get stuck with this destructuring error. I wonder if it's actually useful to force them to encounter it? Maybe if we framed it more deliberately so they're expecting it?

> **Now you _might_ be seeing something! *If not, that's fine, you've followed instructions correctly*.**

This isn't technically true: they should definitely be seeing an error at this point. It might be better to say something like "Now let's try accessing the data—this won't work at first, and we'll find out why after." Then get them to try and see the data isn't there yet.

**Comment by sofiapoh** [comment]
I completely forgot about this annoying part. I'll make a clarification that they'll definitely see an error first. It's probably worth showing it so they get some exposure to what that sort of error may look like? Not sure!

**Comment by oliverjam** [comment]
Yeah I think it's fine as long as we make it clear they'll have an error _before_ we get them to write the code. People often hit the error and get stuck debugging and never read the following section.

It would also be good to have them log the data in the component's render so they can see it is null at first.
